crates/bitwarden-crypto/src/service/mod.rs

```diff
@@ -0,0 +1,197 @@
+use std::sync::{Arc, RwLock};
+
+use crate::{AsymmetricKeyRef, Decryptable, Encryptable, KeyRef, SymmetricKeyRef, UsesKey};
+
+mod context;
+
+mod key_store;
+
+use context::ReadWriteGlobal;
+pub use context::{CryptoServiceContext, ReadOnlyGlobal};
+pub use key_store::create_key_store;
+use key_store::KeyStore;
+
+#[derive(Clone)]
+pub struct CryptoService<SymmKeyRef: SymmetricKeyRef, AsymmKeyRef: AsymmetricKeyRef> {
```

Comment:
**question**: We define a Keys struct which essentially acts as the KeyStore. Would it be a cleaner interface to define a KeyStore trait which both CryptoService and Keys implements, and CryptoService simply forwards calls to the inner store.

Reply:
Initially I think I've tried to do something like that, but it was hard to get a good API when using a trait here and then also trying to be generic over key type / Encryptable implementation.